lib/matplotlib/axes/_axes.py

```diff
@@ -2604,6 +2605,137 @@ def barh(self, y, width, height=0.8, left=None, *, align="center",
                            align=align, **kwargs)
         return patches
 
+    def bar_label(self, container, captions=[], fmt="%g", mode="edge",
+                  padding=None, shifting=0, autoscale=True, **kwargs):
+        """
+        Label a bar plot.
+
+        Adds labels to bars in the given `.BarContainer`.
+
+        Parameters
+        ----------
+        container : `.BarContainer`
+            Container with all the bars and optionally errorbars.
+
+        captions : array-like, optional
+            A list of label texts, that should be displayed. If not given, the
+            label texts will be the data values formatted with *fmt*.
+
+        fmt : str, optional
```

Comment:
```suggestion
        fmt : str, default: '%g'
```
Recently, we've decided to put the defaults up there and leave out `optional` if feasible. See https://matplotlib.org/devdocs/devel/documenting_mpl.html#parameter-type-descriptions.